### PR TITLE
Update SK to 1.0.1

### PR DESCRIPTION
**Pull request by dehoward**
### Motivation and Context

WIP

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: webapi/Plugins/Chat/ChatPlugin.cs [hunk 1]
```diff
@@ -190,22 +185,22 @@ private async Task<string> ExtractAudienceAsync(SKContext context, CancellationT
             );
 
         // Clone the context to avoid modifying the original context variables.
-        var audienceExtractionContext = context.Clone();
-        audienceExtractionContext.Variables.Set("tokenLimit", historyTokenBudget.ToString(new NumberFormatInfo()));
+        var audienceExtractionArguments = new KernelArguments(kernelArguments);
+        kernelArguments.Add("tokenLimit", historyTokenBudget);
 
         var completionFunction = this._kernel.CreateSemanticFunction(
```
Comment (linwenda) [inline webapi/Plugins/Chat/ChatPlugin.cs:191]: Perhaps it's necessary to modify it like this now?
`
_kernel.CreateFunctionFromPrompt(new PromptTemplateConfig(_promptOptions.SystemAudienceExtraction)
        {
            Description = "Complete the prompt.",
            Name = nameof(ChatPlugin)
        });
`

Comment (linwenda) [reply webapi/Plugins/Chat/ChatPlugin.cs:191]: However, when i try to invoke it, I noticed that the current invoke method requires passing a '**kernel**', which seems a bit unusual to me. I'm unsure if this design is intentional or if there might be alternative overloaded methods that also require a '**kernel**'

`var result = await completionFunction.InvokeAsync(_kernel, audienceExtractionArguments, cancellationToken);`

**Comment by UTMark** [comment]
Good morning! I was wondering if this is a feature that is still being developed since it has been a month? I'm really looking forward for this implementation since I think it will resolve the web searcher plugin :)

**Comment by hoaho-vlg** [comment]
Hello development team, Is there any plan to merge this PR, I'm looking forward to experiencing the features of the new version of Semantic kernel 1.0.1

**Comment by philiphuynh-vlg** [comment]
When this pull request will be merged?

**Comment by ezar** [comment]
this PR broke Planner :(

**Comment by glahaye** [comment]
Replaced by #817

**Pull request closed without merging**